### ISSUE TITLE
Add ability to configure needed native-image C/C++ dependencies

### ISSUE THREAD

**Issue by johnynek** [oracle/graal#1557]
native-image needs gcc, glibc and zlib (according to: https://www.graalvm.org/docs/reference-manual/aot-compilation/ ).

It seems to require `gcc` to be on the path:
https://github.com/oracle/graal/blob/f30f2aa2f0fae4eeb21053fa965bd045abf3a805/substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/c/codegen/CCompilerInvoker.java

It would be nice for building tools that use native image to allow passing an argument to the path to `gcc` so that we can more easily build support for graal with bazel (e.g. see: https://github.com/andyscott/rules_graal/issues/8 ). Bazel ideally would like to know the detail s of the c/c++ toolchain and you can query the path to the compiler, but it doesn't just put things on the PATH.

Additionally, it would be nice to have a documented way to pass options to include and link the path to glibc or zlib that we want.

**Comment by renau** [comment]
Any update on this. Needed for bazel

**Comment by olpaw** [comment]
You are right, atm the use of gcc available via PATH  (on non-Windows platforms) is hardcoded into native-image. Also it should be possible to specify default compiler flags. I will add two hosted options to fix this issue.

**Comment by olpaw** [comment]
Fixed in e2de803a606769158fad0e79ca5af04c4751f279
Now there are two additional hosted options for `native-image`:
```
  -H:CCompilerOption=...                       Provide custom C compiler option used for query code compilation. Default: None
  -H:CCompilerPath=...                         Provide custom path to C compiler used for query code compilation and linking. Default: None
```
